test/segment-loader.test.js

```diff
@@ -442,6 +442,42 @@ QUnit.module('SegmentLoader: M2TS', function(hooks) {
       assert.equal(loader.mediaRequests, 1, '1 request');
     });
 
+    QUnit.test('sets the timestampOffset on timeline change but not if startOfSegment is early', function(assert) {
+      let playlist = playlistWithDuration(40);
+      let buffered = videojs.createTimeRanges();
+      let hlsTimestampOffsetEvents = 0;
+
+      loader.on('timestampoffset', () => {
+        hlsTimestampOffsetEvents++;
+      });
+
+      loader.buffered_ = () => buffered;
+
+      loader.playlist(playlist);
+      loader.mimeType(this.mimeType);
+      loader.load();
+      this.clock.tick(1);
+
+      // segment 0
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+      buffered = videojs.createTimeRanges([[0, 10]]);
+      this.updateend();
+      this.clock.tick(1);
+
+      assert.equal(hlsTimestampOffsetEvents, 0,
+        'no hls-timestamp-offset event was fired');
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+      loader.sourceUpdater_.timestampOffset_ = 11;
```

Comment:
what is this line doing?

Reply:
terrible things but it's responsible for making the conditional that was removed pass

Reply:
I think you would want to do this before the updateend or clock tick above 465-466

Reply:
apparently, nothing.